src/drivers/ammo-driver.js

```diff
@@ -1,10 +1,174 @@
+  this.solver = null;
+  this.physicsWorld = null;
+  this.debugDrawer = null;
+
+  this.els = {};
+  this.eventListeners = [];
+  this.collisions = {};
 }
 
 AmmoDriver.prototype = new Driver();
 AmmoDriver.prototype.constructor = AmmoDriver;
 
 module.exports = AmmoDriver;
+
+/* @param {object} worldConfig */
+AmmoDriver.prototype.init = function(worldConfig) {
+  return new Promise(resolve => {
+    AmmoModule().then(result => {
+      Ammo = result;
+      this.epsilon = worldConfig.epsilon || EPS;
+      this.debugDrawMode = worldConfig.debugDrawMode || THREE.AmmoDebugConstants.NoDebug;
+      this.maxSubSteps = worldConfig.maxSubSteps || 4;
+      this.fixedTimeStep = worldConfig.fixedTimeStep || 1 / 60;
+      this.collisionConfiguration = new Ammo.btDefaultCollisionConfiguration();
+      this.dispatcher = new Ammo.btCollisionDispatcher(this.collisionConfiguration);
+      this.broadphase = new Ammo.btDbvtBroadphase();
+      this.solver = new Ammo.btSequentialImpulseConstraintSolver();
+      this.physicsWorld = new Ammo.btDiscreteDynamicsWorld(
+        this.dispatcher,
+        this.broadphase,
+        this.solver,
+        this.collisionConfiguration
+      );
+      this.physicsWorld.setGravity(
+        new Ammo.btVector3(0, worldConfig.hasOwnProperty("gravity") ? worldConfig.gravity : -9.8, 0)
+      );
+      this.physicsWorld.getSolverInfo().set_m_numIterations(worldConfig.solverIterations);
+      resolve();
+    });
+  });
+};
+
+/* @param {Ammo.btCollisionObject} body */
+AmmoDriver.prototype.addBody = function(body, group, mask) {
+  this.physicsWorld.addRigidBody(body, group, mask);
+  this.els[Ammo.getPointer(body)] = body.el;
+};
+
+/* @param {Ammo.btCollisionObject} body */
+AmmoDriver.prototype.removeBody = function(body) {
+  this.physicsWorld.removeRigidBody(body);
+  delete this.els[Ammo.getPointer(body)];
+};
+
+AmmoDriver.prototype.updateBody = function(body) {
+  if (this.els.hasOwnProperty(Ammo.getPointer(body))) {
+    this.physicsWorld.updateSingleAabb(body);
+  }
+};
+
+/* @param {number} deltaTime */
+AmmoDriver.prototype.step = function(deltaTime) {
+  this.physicsWorld.stepSimulation(deltaTime, this.maxSubSteps, this.fixedTimeStep);
+
+  for (const key in this.collisions) {
+    this.collisions[key].colliding = false;
+  }
+
+  let numManifolds = this.dispatcher.getNumManifolds();
+  for (let i = 0; i < numManifolds; i++) {
+    let persistentManifold = this.dispatcher.getManifoldByIndexInternal(i);
+    let numContacts = persistentManifold.getNumContacts();
+    let body0 = persistentManifold.getBody0();
+    let body1 = persistentManifold.getBody1();
+    let key = Ammo.getPointer(body0) + "_" + Ammo.getPointer(body1);
+    let collided = false;
+    for (let j = 0; j < numContacts; j++) {
+      let manifoldPoint = persistentManifold.getContactPoint(j);
+      let distance = manifoldPoint.getDistance();
+      if (Ammo.getPointer(body0) !== Ammo.getPointer(body1) && distance <= this.epsilon) {
```

Comment:
Why do we need to check pointers to see if the bodies are the same thing? Is ammoJS using a pool, and the actual instances are fluid?

Reply:
It seems that it is possible that a persistentManifold can exist where both bodies are the same and I'm not entirely sure why. 
I'm not entirely sure what you're asking in your second question, but while ammo doesn't explicitly pool pointers for btRigidBodies, it's possible for a pointer to be reused if that body is deleted and a new one is created (These pointers are just indexes into the heap allocated for emscripten). 
Ammo _does_ seem to pool pointers used for persistentManifolds (so a pointer to a persistentManifold cannot be used to explicitly refer to a pair of bodies), though I'm not sure if that's what you're asking either.

Reply:
I'm just wondering why we use `Ammo.getPointer(body0) !== Ammo.getPointer(body1)` instead of the more obvious alternative, `body0 !== body1`. If both bodies were the same instance, `getPointer` would have to return the same value both times, right?

Reply:
In the most recent version of this function, those pointers are grabbed before this loop, so this is kind of a moot point. I forget if this was strictly necessary at the time or not though. You're probably right in that it was not necessary to check the pointers specifically.